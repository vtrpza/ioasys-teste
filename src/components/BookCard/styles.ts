import styled, { css } from 'styled-components'

export const Wrapper = styled.section`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  width: 27rem;
  vertical-align: center;
  height: 16rem;
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    margin-left: ${theme.spacings.xsmall};
    margin-bottom: ${theme.spacings.xsmall};
  `}
  cursor: pointer;
  box-shadow: 0px 1px 10px rgba(0, 0, 0, 0.04), 0px 8px 80px rgba(0, 0, 0, 0.09);
`

export const ImageBox = styled.div`
  height: 16rem;
  width: 12rem;
  ${({ theme }) => css`
    padding: ${theme.spacings.xxsmall};
  `}

  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    box-shadow: 0px 1.8px 2.2px rgba(0, 0, 0, 0.017),
      0px 4.3px 5.3px rgba(0, 0, 0, 0.024), 0px 8.1px 10px rgba(0, 0, 0, 0.03),
      0px 14.5px 17.9px rgba(0, 0, 0, 0.036),
      0px 27.2px 33.4px rgba(0, 0, 0, 0.043), 0px 65px 80px rgba(0, 0, 0, 0.06);
  }
`
export const Info = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  max-width: 50%;
  overflow: hidden;
  ${({ theme }) =>
    css`
      padding-left: ${theme.spacings.xxsmall};
    `}
`

export const PrimaryInfo = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  text-align: left;
  ${({ theme }) => css`
    padding: ${theme.spacings.xsmall} 0;
  `}
`

export const SecondaryInfo = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  text-align: left;
  ${({ theme }) => css`
    padding: ${theme.spacings.xxsmall} 0;
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.black};
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.small};
  `}
`

export const Author = styled.h3`
  ${({ theme }) => css`
    color: ${theme.colors.darkPink};
    font-weight: ${theme.font.normal};
    font-size: ${theme.font.sizes.xsmall};
  `}
`

export const Pages = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-weight: ${theme.font.light};
    font-size: ${theme.font.sizes.xsmall};
  `}
`

export const Publisher = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-weight: ${theme.font.light};
    font-size: ${theme.font.sizes.xsmall};
  `}
`

export const Published = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-weight: ${theme.font.light};
    font-size: ${theme.font.sizes.xsmall};
  `}
`
