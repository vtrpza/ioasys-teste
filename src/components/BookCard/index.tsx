import * as S from './styles'

export type BookCardProps = {
  title: string
  author: string
  pages: string
  publisher: string
  published: string
  img: string
}

const BookCard = ({
  title,
  author,
  pages,
  publisher,
  published,
  img
}: BookCardProps) => (
  <S.Wrapper>
    <S.ImageBox>
      <img src={img} alt={title} />
    </S.ImageBox>
    <S.Info>
      <S.PrimaryInfo>
        <S.Title>{title}</S.Title>
        <S.Author>{author}</S.Author>
      </S.PrimaryInfo>
      <S.SecondaryInfo>
        <S.Pages>{pages}</S.Pages>
        <S.Publisher>{publisher}</S.Publisher>
        <S.Published>{published}</S.Published>
      </S.SecondaryInfo>
    </S.Info>
  </S.Wrapper>
)

export default BookCard
