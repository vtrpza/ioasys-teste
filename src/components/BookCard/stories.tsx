import { Story, Meta } from '@storybook/react/types-6-0'
import BookCard from '.'

const props = {
  id: '2',
  title: '23423',
  authors: ['te', 'te'],
  pageCount: '12312',
  publisher: '12312',
  published: '12312',
  category: '12312',
  language: '12312',
  imageUrl: './img/Book 1.png'
}

export default {
  title: 'BookCard',
  component: BookCard
} as Meta

export const Default: Story = (args) => <BookCard {...props} {...args} />
