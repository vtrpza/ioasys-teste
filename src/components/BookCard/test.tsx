import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import BookCard from '.'

const props = {
  title: 'Livro 1',
  author: 'Vitor Pouza',
  pages: '49',
  publisher: 'PZA Inc',
  published: '21 abril 2022',
  img: './img/Book 1.png'
}

describe('<BookCard/>', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<BookCard {...props} />)

    expect(
      screen.getByRole('heading', { name: /Livro 1/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
