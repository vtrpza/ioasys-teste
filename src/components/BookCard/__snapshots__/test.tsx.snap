// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookCard/> should render the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 27rem;
  vertical-align: center;
  height: 16rem;
  background-color: #FFFFFF;
  margin-left: 1.6rem;
  margin-bottom: 1.6rem;
  cursor: pointer;
  box-shadow: 0px 1px 10px rgba(0,0,0,0.04),0px 8px 80px rgba(0,0,0,0.09);
}

.c1 {
  height: 16rem;
  width: 12rem;
  padding: 0.8rem;
}

.c1 img {
  width: 100%;
  height: 100%;
  object-fit: cover;
  box-shadow: 0px 1.8px 2.2px rgba(0,0,0,0.017),0px 4.3px 5.3px rgba(0,0,0,0.024),0px 8.1px 10px rgba(0,0,0,0.03),0px 14.5px 17.9px rgba(0,0,0,0.036),0px 27.2px 33.4px rgba(0,0,0,0.043),0px 65px 80px rgba(0,0,0,0.06);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-left: 0.8rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: left;
  padding: 1.6rem 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: left;
  padding: 2.4rem 0;
}

.c4 {
  color: #333333;
  font-weight: 500;
  font-size: 1.4rem;
}

.c5 {
  color: #AB2680;
  font-weight: 400;
  font-size: 1.2rem;
}

.c7 {
  color: #999999;
  font-weight: 300;
  font-size: 1.2rem;
}

.c8 {
  color: #999999;
  font-weight: 300;
  font-size: 1.2rem;
}

.c9 {
  color: #999999;
  font-weight: 300;
  font-size: 1.2rem;
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Livro 1"
      src="./img/Book 1.png"
    />
  </div>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <h2
        class="c4"
      >
        Livro 1
      </h2>
      <h3
        class="c5"
      >
        Vitor Pouza
      </h3>
    </div>
    <div
      class="c6"
    >
      <p
        class="c7"
      >
        49
      </p>
      <p
        class="c8"
      >
        PZA Inc
      </p>
      <p
        class="c9"
      >
        21 abril 2022
      </p>
    </div>
  </div>
</section>
`
